### PR TITLE
[ACS] format default dns_name_prefix

### PR DESCRIPTION
**Pull request by rjtsdl**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by rjtsdl** [comment]
fix #4116

**Comment by rjtsdl** [comment]
@anhowe TAL

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def _get_feature_in_preview_message():
 # some admin names are prohibited in acs, such as root, admin, etc. Because we have no control on the orchestrators, so default to a safe name.
 register_cli_argument('acs', 'admin_username', options_list=('--admin-username',), default='azureuser', required=False)
 register_cli_argument('acs', 'api_version', options_list=('--api-version',), required=False, help=_get_feature_in_preview_message() + 'Use API version of ACS to perform az acs operations. Available options: 2017-01-31, 2017-07-01. Default to use the latest version for the location')
-register_cli_argument('acs', 'dns_name_prefix', options_list=('--dns-prefix', '-d'))
+register_cli_argument('acs', 'dns_name_prefix', options_list=('--dns-prefix', '-d'), help='default use the format of <clustername>-<resourcegroupname>-<subid>, will trim the length and replace sensitive charactors if needed')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py]: typo: characters

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 2]
```diff
@@ -483,7 +484,11 @@ def acs_create(resource_group_name, deployment_name, name, ssh_key_value, dns_na
     subscription_id = _get_subscription_id()
     if not dns_name_prefix:
         # Use subscription id to provide uniqueness and prevent DNS name clashes
-        dns_name_prefix = '{}-{}-{}'.format(name, resource_group_name, subscription_id[0:6])
+        name_part = re.sub('[^A-Za-z0-9-]', '', name)[0:10]
+        if not name_part[0].isalpha():
+            name_part = (str('a') + name_part)[0:10]
+        resource_group_part = re.sub('[^A-Za-z0-9-]', '', resource_group_name)[0:16]
+        dns_name_prefix = '{}-{}-{}'.format(name_part, resource_group_part, subscription_id[0:6])
```
Comment (tjprescott) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: I'd recommend extracting this logic to method and adding a few unit tests.

**Comment by rjtsdl** [comment]
@tjprescott , please help merge if you approve :)

**Pull request merged**